packages/protocol-http/src/extensions/httpExtensionConfiguration.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import {
+  getHttpHandlerConfiguration,
+  resolveHttpHandlerRuntimeConfig as _resolveHttpHandlerRuntimeConfig,
+} from "./http";
+
+/**
+ * @internal
+ */
+export interface HttpHandlerExtensionConfiguration<HandlerConfig extends object = Record<string, unknown>> {
+  setHttpHandler(handler: HttpHandler<HandlerConfig>): void;
+  httpHandler(): HttpHandler<HandlerConfig>;
+  updateHttpClientConfig(key: keyof HandlerConfig, value: HandlerConfig[typeof key]): void;
+  httpHandlerConfigs(): HandlerConfig;
+}
+
+/**
+ * @internal
+ */
+export type HttpHandlerExtensionConfigType = Parameters<typeof getHttpHandlerConfiguration>[0];
+
+/**
+ * @internal
+ *
+ * Helper function to resolve default extension configuration from runtime config
+ */
+export const getHttpHandlerExtensionConfiguration = (runtimeConfig: HttpHandlerExtensionConfigType) => {
+  return {
+    ...getHttpHandlerConfiguration(runtimeConfig),
+  };
+};
+
+/**
+ * @internal
+ *
+ * Helper function to resolve runtime config from default extension configuration
+ */
+export const resolveHttpHandlerRuntimeConfig = (config: HttpHandlerExtensionConfiguration) => {
```

Comment:
why is this function needed? it only performs an object copy?

Reply:
Initially there was plan to add authentication related logic inside resolveHttpHandlerRuntimeConfig. I will remove the duplicated function for now.